apps/extension/src/popup.ts

```diff
@@ -6,14 +6,22 @@ import type {
   InternalResponse,
 } from '@penumbra-zone/types/src/internal-msg/shared';
 import { Code, ConnectError } from '@connectrpc/connect';
+import { isChromeResponderDroppedError } from '@penumbra-zone/types/src/internal-msg/chrome-error';
 
 export const popup = async <M extends PopupMessage>(
   req: PopupRequest<M>,
 ): Promise<PopupResponse<M>> => {
   await spawnPopup(req.type);
   // We have to wait for React to bootup, navigate to the page, and render the components
-  await new Promise(resolve => setTimeout(resolve, 1000));
-  return chrome.runtime.sendMessage<InternalRequest<M>, InternalResponse<M>>(req);
+  await new Promise(resolve => setTimeout(resolve, 800));
+  return chrome.runtime.sendMessage<InternalRequest<M>, InternalResponse<M>>(req).catch(e => {
+    if (isChromeResponderDroppedError(e))
+      return {
+        type: req.type,
+        data: null,
+      };
+    else throw e;
+  });
```

Comment:
I get what's happening here, but not how it fits into the PR as a whole

Reply:
this pr is addressing cross-document error propagation.

the closing of the popup document surfaces as an error, but it is not a failure, it is a meaningful response. handling the termination of a document from within the document is complex, and we shouldn't reimplement it for every kind of popup, so it's handled generally for all popups.